src/library/commissioner_safe.cpp

```diff
@@ -562,10 +562,7 @@ CommissionerSafe::AsyncRequest CommissionerSafe::PopAsyncRequest()
 CommissionerSafe::EventBaseHolder::EventBaseHolder()
     : mEventBase(event_base_new())
 {
-    if (mEventBase == nullptr)
```

Comment:
exception not allowed even when memory error ?

Reply:
I think so, we should avoid `throw` when possible. Here we want to just terminate the program, and `VerifyOrDie` can do this for us.

Reply:
I think is not proper to just **EXIT** the process when creating `CommissionerSafe` failed. 
The user of `CommissionerSafe` would expect following code to return nil when failed ?

```c++
  CommissionerSafe *c = new CommissionerSafe();
```

Reply:
Changed to check the failure in `Init`, please review @simonlingoogle